packages/site/src/hooks/MetamaskContext.tsx

```diff
@@ -1,71 +1,23 @@
 import type { MetaMaskInpageProvider } from '@metamask/providers';
-import type { Dispatch, ReactNode, Reducer } from 'react';
-import { createContext, useEffect, useReducer, useState } from 'react';
+import type { ReactNode } from 'react';
+import { createContext, useContext, useEffect, useState } from 'react';
 
-import type { Snap } from '../types';
-import { getSnapsProvider, getSnap, isFlask } from '../utils';
+import { getSnapsProvider } from '../utils';
 
-export type MetamaskState = {
-  snapsDetected: boolean;
-  isFlask: boolean;
-  installedSnap?: Snap;
-  error?: Error;
-};
-
-const initialState: MetamaskState = {
-  snapsDetected: false,
-  isFlask: false,
+type MetaMaskContextType = {
+  provider: MetaMaskInpageProvider | null;
+  error: Error | undefined;
```

Comment:
Should we use `null` for consistency?

```suggestion
  error: Error | null;
```